diff --git a/src-tauri/Cargo.toml b/src-tauri/Cargo.toml
index c998577..c92139e 100644
--- a/src-tauri/Cargo.toml
+++ b/src-tauri/Cargo.toml
@@ -35,3 +35,3 @@ tauri-plugin-dialog = "2.7.1"
 # when `tauri dev` runs it is executed with `cargo run --no-default-features` if `devPath` is an URL
-default = [ "custom-protocol", "updater" ]
+default = [ "custom-protocol" ]
 # this feature is used used for production builds where `devPath` points to the filesystem
 